aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1332,7 +1307,19 @@ impl AptosVM {
         )?;
 
         Self::validate_waypoint_change_set(&change_set, log_context)?;
-        self.read_change_set(resolver.as_executor_view(), &change_set)?;
+        // All Move executions satisfy the read-before-write property. Thus we need to read each
+        // access path that the write set is going to update.
+        let executor_view = resolver.as_executor_view();
+        for state_key in change_set.module_write_set().keys() {
+            executor_view
+                .get_module_state_value(state_key)
+                .map_err(|_| VMStatus::error(StatusCode::STORAGE_ERROR, None))?;
+        }
+        for state_key in change_set.resource_write_set().keys() {
+            executor_view
+                .get_resource_state_value(state_key, None)
```

Comment:
I realised that aggregator V1 will actually go down this path: when we resolve it using AggregatorView we redirect to ResourceView anyway. But what about resource groups? If I have a blob which is group it's write will go unnoticed?

Reply:
hm, but don't I have to iterate over aggregator_v1_write_set as well?

It's a really good question for the new resource group write set. I can dispatch a read for each resource in there?

but there is even another good question, what happens for aggregator v2? depending on how this write-set is generated, will it require the replacement or not? (+@zekun000 @igor-aptos )